tests/functional/single_machine_outputs.nix

```diff
@@ -0,0 +1,16 @@
+{
+  resources = {
+    output.thing = {
+      script = ''
+        #!/bin/sh
+        echo -n '"12345"'
+      '';
+    };
+  };
+  machine = {resources, ...} : {
+    imports = [ <nixpkgs/nixos/modules/profiles/minimal.nix> ];
+    deployment = {
+      keys."secret.key".text = resources.output.thing.value;
```

Comment:
This requires the 3 evaluations you added, so I suggest another example/test ?

Reply:
Added a new test.